pyaedt/twinbuilder.py

```diff
@@ -309,5 +312,258 @@ def set_sim_setup_parameter(self, var_str, expression, analysis_name="TR"):
+            2D model depth specified as value with unit.
+            To be provided if design is Q2D.
+            The default value is ``1meter``
+        maximum_order : float, optional
+            The Maximum Order value specifies the highest order state space
+            system that you can choose while fitting to represent the system behavior.
+            A lower order may lead to less accuracy but faster simulation.
+            The default value is ``10000``.
+        error_tolerance : float, optional
+            Error Tolerance sets the error tolerance for S-Matrix fitting.
+            The default value is ``0.005``.
+        z_ref : float, optional
+            Sets the value of the Z (ref) in ohms.
+            The default value is ``50``.
+        state_space_dynamic_link_type : str, optional
+            Q3D state space dynamic link type.
+            Possible options are:
+                - ``S`` for S parameters link type.
+                - ``RLGC`` for RLGC Parameters link type.
+                - ``EQ`` for Equivalent Circuit.
+            The default value is ``RLGC``.
+        component_name : str, optional
+            Component name.
+            If not provided a generic name with prefix "SimpQ3DData" will be given.
+        save_project : bool, optional
+            Save project after use.
+            The default value is ``True``.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+
+        References
+        ----------
+        >>> oComponentManager.AddDynamicNPortData
+
+        Examples
+        --------
+        Create an instance of Twin Builder.
+
+        >>> from pyaedt import TwinBuilder
+        >>> tb = TwinBuilder()
+
+        Add a Q2D dynamic link component.
+
+        >>> tb.add_q3d_dynamic_component("Q2D_ArmouredCableExample",
+        ...                             "2D_Extractor_Cable",
+        ...                             "MySetupAuto",
+        ...                             "sweep1",
+        ...                             "Original",
+        ...                             "100mm")
+        >>> tb.release_desktop()
+        """
+        dkp = self.desktop_class
+        is_loaded = False
+        app = None
+        if os.path.isfile(source_project):
+            if os.path.exists(source_project):
+                project_path = source_project
+                project_name = os.path.splitext(os.path.basename(source_project))[0]
+                if project_name in dkp.project_list():
+                    app = dkp[[project_name, source_design_name]]
+                else:
+                    app = dkp.load_project(project_path, source_design_name)
+                    is_loaded = True
+        elif source_project in self.project_list:
+            project_name = source_project
+            project_path = os.path.join(self.project_path, project_name + ".aedt")
+            project_path = project_path.replace("/", "\\")
```

Comment:
Doesn't that break with linux ?

Reply:
I don't know honestly

Reply:
I see that it is different from what we saw together. 
@SMoraisAnsys the recording shows that in windows the backslash "\" are escaped in the string itself, so they should be "\\\\" four of them in python. 
Maybe everything works also with a single slash "/", in this case no escape is necessary and we don't have to worry for Linux. 
@gmalinve can you try if it works with a single forward slash "/" in the path? Thanks

Reply:
@Alberto-DM @gmalinve There might be a misunderstanding here, I just wanted to pinpoint the replace call. The call `os.path.join` is compatible with both windows and linux so I though that using replace would break the linux compatibility. However, if there is something specific to `self.project_path` that might need such a modification, it might be interesting to only perform the replace on `self.project_path`. Forgive my ignorance if my remark is unfounded.

```python
# In windows
path = 'C:\\Users\\ansys'
res = os.path.join(path, 'some_dir') # res would be equal to 'C:\\Users\\ansys\\some_dir'
# In linux
path = '/home/ansys'
res = os.path.join(path, 'some_dir') # res would be equal to '/home/ansys/some_dir'
```

Reply:
@SMoraisAnsys @Alberto-DM tests pass anyway so I'll remove the replace method and leave it as is. Thanks both for feedback! :)